src/Microsoft.DocAsCode.Build.Engine/DocumentBuilder.cs

```diff
@@ -427,45 +431,42 @@ private static void BuildArticle(IDocumentProcessor processor, HostService hostS
                 maxParallelism);
         }
 
-        private static void Postbuild(IDocumentProcessor processor, HostService hostService)
+        private static void Postbuild(HostService hostService)
         {
             RunBuildSteps(
-                processor.BuildSteps,
+                hostService.Processor.BuildSteps,
                 buildStep =>
                 {
-                    Logger.LogVerbose($"Processor {processor.Name}, step {buildStep.Name}: Postprocessing...");
+                    Logger.LogVerbose($"Processor {hostService.Processor.Name}, step {buildStep.Name}: Postprocessing...");
                     using (new LoggerPhaseScope(buildStep.Name))
                     {
                         buildStep.Postbuild(hostService.Models, hostService);
                     }
                 });
         }
 
-        private IEnumerable<ManifestItemWithContext> ExportManifest(InnerBuildContext buildContext, DocumentBuildContext context)
+        private IEnumerable<ManifestItemWithContext> ExportManifest(HostService hostService, DocumentBuildContext context)
         {
-            var hostService = buildContext.HostService;
-            var processor = buildContext.Processor;
-            var templateProcessor = buildContext.TemplateProcessor;
```

Comment:
I don't want `build context` and `inner build context`, and I think no one know the different between them.
The `inner build context` is a named tuple for `host service`, `document processor` and `template processor`. And `host service` is an internal class, so merge them into that.
And now `host service` means the context for a document processor, and `IHostService` is a set of services providing for plug-ins.